zjit/src/codegen.rs

```diff
@@ -2002,7 +2013,7 @@ c_callable! {
             // function_stub_hit_body() may allocate and call gc_validate_pc(), so we always set PC.
             let iseq_call = unsafe { Rc::from_raw(iseq_call_ptr as *const IseqCall) };
             let iseq = iseq_call.iseq.get();
-            let pc = unsafe { rb_iseq_pc_at_idx(iseq, 0) }; // TODO: handle opt_pc once supported
+            let pc = unsafe { rb_iseq_pc_at_idx(iseq, iseq_call.num_optionals_passed) };
```

Comment:
You need to convert `num_optionals_passed` to `insn_idx` to use this `rb_iseq_pc_at_idx` API.

```suggestion
            let insn_idx = jit_entry_insns(iseq)[iseq_call.num_optionals_passed];
            let pc = unsafe { rb_iseq_pc_at_idx(iseq, insn_idx) };
```